docs/java-your-first-app.md

```diff
@@ -0,0 +1,163 @@
+
+This tutorial is aimed at developers who are new to Temporal and have some basic knowledge of Java. By following this tutorial you will learn how to run a Temporal Workflow application using the [Temporal Java SDK](https://github.com/temporalio/java-sdk) and you will also witness the fundamentally resilient nature of a Workflow, getting just a taste of the value that Temporal can provide.
+
+## Project setup
+
+![Tool box](/img/docs/repair-tools.png)
+
+First, make sure you have the [tutorial prerequisites](/docs/java-sdk-tutorial-prerequisites) setup.
+
+Next, download the ['java-project-template'](https://github.com/temporalio/java-project-template) by downloading it as a zip or by [creating a new repository](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-from-a-template#creating-a-repository-from-a-template) in your own Github account and then cloning the repo to the location of your choice.
+
+The last setup step is to build the project. If you open it with [IntelliJ](https://www.jetbrains.com/idea/) the project will build automatically. If you prefer to run programs from your terminal, make sure you have [Gradle](https://gradle.org/install/) installed and run the Gradle build command from the root of the project:
+
+```
+./gradlew build
+```
+
+Once your project has built, you are ready to go.
+
+## Application overview
+
+![Tool box](/img/docs/workflow.png)
+
+The Java project template mimics a "money transfer" application in order to give you the minimum elements needed to both get started building your own application as well as understand some of the value that Temporal gives you right out of the box. The project includes a predefined [Workflow](/docs/workflows) that orchestrates the execution of an account object's methods `withdraw()` and `deposit()` to mock the transfer of money from one account to another. Temporal calls such methods [Activities](/docs/activities).
+
+The first thing you will do is to call on the Temporal server to start the transfer. Temporal will then start tracking the progress of the Workflow execution. You will then run a Worker whose sole job is to execute the functions of the Workflow and communicate the results back to the Temporal server.
+
+![High level project design](/img/docs/temporal-high-level-application-design.png)
+
+### The Workflow
+
+With Temporal, when you "start" a Workflow, what you are really doing is telling the Temporal server about code that needs to be run and to start tracking its state. The server then breaks up the code into Tasks and sends each Task to a Task Queue. The code below represents three Tasks, one for the Workflow itself, and one for each of the Activity methods.
```

Comment:
Ok so this is what actually happens.

1. On worker you listen to a task queue
2. On worker you register implementations of workflows and activities you want to handle on that worker. It's very important to note that in Java the "id" of workflow is actually just the workflow method name.
3. Send start signal to Temporal server (frontend service), the start includes the task queue name and workflow id
4. Task queue has task added to start and worker picks up that task
5. Worker matches workflow id to registered implementation id and starts running code
6. When worker reaches a stateful change it sources the event back to the history service
7. The next task is added to the queue and a new worker picks it up
8. so on and so forth.

Reply:
OK, so I rewrote this bit - and tried to make it accurate but simple.